rpc/rpcserver/server.go

```diff
@@ -2340,6 +2347,63 @@ func (s *loaderServer) FetchMissingCFilters(ctx context.Context, req *pb.FetchMi
 	return &pb.FetchMissingCFiltersResponse{}, nil
 }
 
+func (s *loaderServer) SpvSync(req *pb.SpvSyncRequest, svr pb.WalletLoaderService_SpvSyncServer) error {
+
+	wallet, ok := s.loader.LoadedWallet()
+	if !ok {
+		return status.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+
+	if req.DiscoverAccounts && len(req.PrivatePassphrase) == 0 {
+		return status.Errorf(codes.InvalidArgument, "private passphrase is required for discovering accounts")
+	}
+
+	if req.DiscoverAccounts {
+		lock := make(chan time.Time, 1)
+		defer func() {
+			lock <- time.Time{}
+			zero.Bytes(req.PrivatePassphrase)
+		}()
+		err := wallet.Unlock(req.PrivatePassphrase, lock)
+		if err != nil {
+			return translateError(err)
+		}
+	}
+
+	addr := &net.TCPAddr{IP: net.ParseIP("::1"), Port: 0}
+	amgrDir := filepath.Join(s.appData, wallet.ChainParams().Name)
+	amgr := addrmgr.New(amgrDir, net.LookupIP) // TODO: be mindful of tor
+	lp := p2p.NewLocalPeer(wallet.ChainParams(), addr, amgr)
+
+	ntfns := &spv.NtfnsCallbacks{
+		SyncUpdated: func(sync bool) {
+			resp := &pb.SpvSyncResponse{Synced: sync}
+			// TODO: Add some kind of logging here.  Do nothing with the error
+			// for now. Could be nice to see what happened, but not super
+			// important.
+			_ = svr.Send(resp)
+		},
+	}
+	syncer := spv.NewSyncer(wallet, lp, ntfns)
+	if len(req.SpvConnect) > 0 {
+		spvConnect, err := cfgutil.NormalizeAddress(req.SpvConnect, s.activeNet.Params.DefaultPort)
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "SPV Connect address invalid: %v", err)
+		}
+		spvConnects := make([]string, 1)
+		spvConnects[0] = spvConnect
+		syncer.SetPersistantPeers(spvConnects)
+	}
+	wallet.SetNetworkBackend(syncer)
+	s.loader.SetNetworkBackend(syncer)
+
+	err := syncer.Run(svr.Context())
+	if err != nil {
+		return status.Errorf(codes.FailedPrecondition, "SPV synchronization ended: %v", err)
```

Comment:
this is wrong, it will return a FailedPrecondition error when the context is canceled.